### DISCUSSION TITLE
Production Ready?

### DISCUSSION THREAD

**Discussion by Rshep3087** [riverqueue/river#208] [General]
I am interested if you are aware of any instances of River being used in a production workload? I want to move away from background jobs being managed with Go concurrency primitives and River seems like a solid choice. 

I have it running in our lower environments right now to get an idea of how it performs, but it would be helpful to know if others are having success with it. Or do you recommend holding off for production workloads until the library becomes more stable.

**Comment by brandur** [comment]
Hi @Rshep3087,

Firstly, we'd love for others to chime in here if they've had luck one way or the other running River in production. I'd guess that by now some people are based on traffic on this repository, but it's hard to tell for sure.

I'll say about it that River's still a relatively new project, so we can't yet guarantee perfect production stability, but we have put a lot of effort into a comprehensive test suite to route out bugs. A few bugs have been reported since our initial release, but we so far have a good track record of getting them fixed quickly.

And again, if anyone has prod River testimonials, please chime in.

**Reply by Rshep3087** [reply]
Hey @brandur, thanks for the response. Hopefully we can get some individuals to chime in. I am going to let it run in our lower environments for now and hopefully run some loads on it at some point. If everything looks good, maybe we will be able to put it into production. I will keep this thread updated if we do.

**Reply by Rshep3087** [reply]
following up, we have been running River for one of our background jobs for about a week. I just needed needed to add a subcommand to output a table to see a snapshot of jobs. We are seeing a slow memory that was brought up in https://github.com/riverqueue/river/issues/239, but our we are not handling as many jobs so it is a slow leak.

<img width="1377" alt="Screenshot 2024-02-29 at 3 04 33 PM" src="https://github.com/riverqueue/river/assets/19920342/9d1a202c-9a46-41d5-9236-93414f460456">

**Reply by brandur** [reply]
Hi @Rshep3087, thanks for the report.

From today: #243 brought in a fix for a context leak, and we've cut [v0.0.24](https://github.com/riverqueue/river/releases/tag/v0.0.24) to correct it. In our tests, memory now seems stable, so we think that was the last one, but will keep an eye on things just in case.

**Comment by leaanthony** [comment]
Keen to hear if there are any others running in Production since last year. My initial load tests have been really good but having knowledge around limitations and boundaries e.g. at what stage does the postgres connection saturate, what are good scaling strategies, etc.
